compiler/rustc_session/src/options.rs

```diff
@@ -150,6 +151,9 @@ top_level_options!(
 
         target_triple: TargetTriple [TRACKED],
 
+        /// Effective logical environment used by `env!`/`option_env!` macros
+        logical_env: FxHashMap<String, String> [TRACKED],
```

Comment:
Hm, this doesn't look great, this is a hash map that we're iterating over. Can you instead use an `FxIndexMap` or `Vec<(_, _)>` for the env vars to ensure a consistent iteration order?

Reply:
Sure.